frontend/src/lib/components/SelectBox.tsx

```diff
@@ -246,25 +260,53 @@ export function SelectUnit({
         }
     }
 
+    const isRowLoaded = ({ index }: { index: number }): boolean => {
+        return Boolean(flattenedData[index])
+    }
+
+    const loadableRowCount = Object.values(loadableItems).reduce(
+        (total, { totalCount: count = 0 }) => (total += count),
+        0
+    )
+    const staticRowCount = Object.values(staticItems).reduce(
+        (total, { dataSource: { length } }) => (total += length),
+        0
+    )
+    const rowCount = loadableRowCount + staticRowCount
+
+    const loadMoreRows = async ({ startIndex, stopIndex }: Record<string, number>): Promise<SelectedItem[]> => {
+        // This function should identify which SelectBoxItem's loader should be called (if available) in order to fetch new rows.
+        console.log('loadMoreRows', { startIndex, stopIndex })
+        return [] // TODO
```

Comment:
To actually see if this works, instead of `return []` why not just wait a second and then return an array of random objects of length `stopIndex - startIndex`? It'll be quick to add, and will already show the interface working (or not).